substrate/network/src/test/mod.rs

```diff
@@ -172,40 +172,38 @@ impl Peer {
 		let mut nonce = 0;
 		if with_tx {
 			self.generate_blocks(count, |builder| {
-				let tx = test_client::runtime::Transaction {
-					from: Keyring::Alice.to_raw_public(),
-					to: Keyring::Alice.to_raw_public(),
-					amount: 1,
-					nonce: nonce,
+				let xt = Extrinsic {
+					signed: Keyring::Alice.to_raw_public().into(),
+					function: Call { to: Keyring::Alice.to_raw_public().into(), amount: 1 },
```

Comment:
Why the change to the extrinsic format? A desire to allow misbehaviour reports in there? Can't we just make it generic and leave unimplemented (or implemented in some other way) for the test chain? I don't want to overspecialise with an assumption that each chain will always have a `Call` in there - that's kind of the point of `test_chain` having a different xt format.